kpatch-build/kpatch-build

```diff
@@ -1021,6 +1021,26 @@ if [[ "$USE_KLP" -eq 1 ]]; then
 	check_pipe_status create-klp-module
 fi
 
+# Check that final module does not reference symbols with different version
+# than the target kernel
+KP_MOD_VALID=true
+# shellcheck disable=SC2086
+while read -ra mod_symbol; do
+    if [[ ${#mod_symbol[@]} -lt 2 ]]; then
+        continue
+    fi
+
+    # First column of the kernel_entry should match the CRC
+    kernel_entry=$(grep "\s${mod_symbol[1]}\s" "$TEMPDIR/Module.symvers")
+    if [[ -n "$kernel_entry" && ! "$kernel_entry" =~ "${mod_symbol[0]}"[[:blank:]] ]]; then
```

Comment:
So, there are two things here.

My approach first look if there is an entry for the same symbol in the new symvers file and if there is, it checks that the CRCs are the same.

This means it doesn't complain when a symbol is not in the new file. But since functions shouldn't get removed, I guess we can assume that if a symbol entry exists in the old file, it should exist in the new one as well. So, unless told otherwise I can try to lookup a line matches crc + symbol version in one go.

As for weirdness, pattern matching in bash is always a mess. I think I'll go with the grep approach as I think it is the softest one for the brain.

Reply:
Anytime I need to manipulate columns, I think of reaching for awk since it provides the built-in `$n` variables.  But it's not for everybody as it reads terse and makes perl look readable :)

I missed the not-in-new-file part of your search and I think keeping it makes more sense than relying on the fact that functions shouldn't be removed.

My only remaining suggestion would be to see if using grep's `-w` word mode and/or word boundaries for the bash regex.  Those are both built-in features that support exactly what you're trying to do manually here.  Adding the space(s) in the patterns initially made me think they were used to dictate which column to match.

Reply:
Adding the spaces manually vs `-w` option allows to tell "this pattern should only be on the first column" or "this pattern should be in one of the inner columns". But then since I'm taking the columns into account you might be right in that the right way to do it is awk.
I'll give awk a go!

Reply:
Also, investigating the `-w` option for the other option. `-w` isn't exactly what we need here.
E.g.:
`grep -w crc8` will match `crc8` but also `lib/crc8` (which exists as a namespace) because `\` counts as a word separator.  With `"\s${sym}\s"` we are sure to consider only the symbol.